[Bug] - Testnet Crash node when call RPC get key from payment address

panic: runtime error: index out of range

goroutine 302318 [running]:
github.com/incognitochain/incognito-chain/wallet.deserialize(0xc000cd4540, 0x21, 0x30, 0x21, 0x30, 0x0)
	/root/go/src/github.com/incognitochain/incognito-chain/wallet/hdwallet.go:172 +0x7f2
github.com/incognitochain/incognito-chain/wallet.Base58CheckDeserialize(0xc002658f80, 0x33, 0xc000c00320, 0xc000e50260, 0x1e)
	/root/go/src/github.com/incognitochain/incognito-chain/wallet/hdwallet.go:198 +0x87
github.com/incognitochain/incognito-chain/rpcserver/rpcservice.GetKeySetFromPaymentAddressParam(0xc002658f80, 0x33, 0xc0007ae630, 0x1, 0x1, 0xc002b3bb00)
	/root/go/src/github.com/incognitochain/incognito-chain/rpcserver/rpcservice/commonservice.go:116 +0x39
github.com/incognitochain/incognito-chain/rpcserver.(*HttpServer).handleGetPublicKeyFromPaymentAddress(0xc000d2d200, 0x1116120, 0xc000c00320, 0xc002b54180, 0xc0009c8e90, 0xc000f92c40, 0x0)
	/root/go/src/github.com/incognitochain/incognito-chain/rpcserver/http_wallet.go:356 +0x79
github.com/incognitochain/incognito-chain/rpcserver.(*HttpServer).ProcessRpcRequest(0xc000d2d200, 0x1699a60, 0xc001bc8000, 0xc000a07800, 0x14d4701)
	/root/go/src/github.com/incognitochain/incognito-chain/rpcserver/http.go:259 +0x109e
github.com/incognitochain/incognito-chain/rpcserver.(*HttpServer).handleRequest.func2(0xc000d2d200, 0x1699a60, 0xc001bc8000, 0xc000a07800, 0x1, 0xc002b54120)
	/root/go/src/github.com/incognitochain/incognito-chain/rpcserver/http.go:164 +0x52
created by github.com/incognitochain/incognito-chain/rpcserver.(*HttpServer).handleRequest
	/root/go/src/github.com/incognitochain/incognito-chain/rpcserver/http.go:163 +0x2ee

--- a/wallet/hdwallet.go
+++ b/wallet/hdwallet.go
@@ -152,6 +152,9 @@ func (key *KeyWallet) Base58CheckSerialize(keyType byte) string {
 // it returns KeySet just contain corresponding key
 func deserialize(data []byte) (*KeyWallet, error) {
 	var key = &KeyWallet{}
+	if len(data) < 2 {
+		return nil, NewWalletError(InvalidKeyTypeErr, nil)
+	}
 	keyType := data[0]
 	if keyType == PriKeyType {
 		key.Depth = data[1]
@@ -169,6 +172,9 @@ func deserialize(data []byte) (*KeyWallet, error) {
 		copy(key.KeySet.PaymentAddress.Tk[:], data[3+apkKeyLength:3+apkKeyLength+pkencKeyLength])
 	} else if keyType == ReadonlyKeyType {
 		apkKeyLength := int(data[1])
+		if len(data) > apkKeyLength+3 {
+			return nil, NewWalletError(InvalidKeyTypeErr, nil)
+		}
 		skencKeyLength := int(data[apkKeyLength+2])
 		key.KeySet.ReadonlyKey.Pk = make([]byte, apkKeyLength)
 		key.KeySet.ReadonlyKey.Rk = make([]byte, skencKeyLength)

--- a/wallet/hdwallet.go
+++ b/wallet/hdwallet.go
@@ -172,7 +172,7 @@ func deserialize(data []byte) (*KeyWallet, error) {
 		copy(key.KeySet.PaymentAddress.Tk[:], data[3+apkKeyLength:3+apkKeyLength+pkencKeyLength])
 	} else if keyType == ReadonlyKeyType {
 		apkKeyLength := int(data[1])
-		if len(data) > apkKeyLength+3 {
+		if len(data) < apkKeyLength+3 {
 			return nil, NewWalletError(InvalidKeyTypeErr, nil)
 		}
 		skencKeyLength := int(data[apkKeyLength+2])

